src/test/groovy/unit/org/pih/warehouse/requisition/RequisitionItemSpec.groovy

```diff
@@ -13,88 +16,346 @@ import spock.lang.Unroll
+        given:
+        RequisitionItem requisitionItem = Spy(RequisitionItem) {
+            totalQuantityCanceled() >> quantityCanceled
+            totalQuantity() >> quantity
+        }
+
+        requisitionItem.modificationItem >> modificationItem
+        requisitionItem.substitutionItem >> substitutionItem
+        requisitionItem.requisitionItems >> requisitionItems
+
+        when:
+        Boolean isRequisitionCanceled = requisitionItem.isCanceled()
+
+        then:
+        isCanceled == isRequisitionCanceled
+
+        where:
+        quantityCanceled | quantity | modificationItem      | substitutionItem      | requisitionItems        || isCanceled
+        3                | 3        | null                  | null                  | null                    || true
+        3                | 2        | null                  | null                  | null                    || false
+        3                | 2        | Mock(RequisitionItem) | null                  | null                    || false
+        3                | 2        | null                  | Mock(RequisitionItem) | null                    || false
+        3                | 2        | null                  | null                  | [Mock(RequisitionItem)] || false
+    }
+
+    void "RequisitionItem.isCanceledDuringPick() should return: #isCanceled when requisition status is #status and quantityPicked of modificationItem is #modificationItemQuantityPicked"() {
+        given:
+        RequisitionItem modificationItem = Stub(RequisitionItem) {
+            calculateQuantityPicked() >> modificationItemQuantityPicked
+        }
+
+        domain.modificationItem = modificationItem
+        domain.requisition.status = status
+
+        when:
+        Boolean isCanceledDuringPick = domain.isCanceledDuringPick()
+
+        then:
+        isCanceledDuringPick == isCanceled
+
+        where:
+        status                      | modificationItemQuantityPicked || isCanceled
+        RequisitionStatus.PICKING   | 0                              || true
+        RequisitionStatus.ISSUED    | 0                              || true
+        RequisitionStatus.VERIFYING | null                           || false
+        RequisitionStatus.PICKING   | null                           || false
+    }
+
+    void "RequisitionItem.isSubstitute() should return: #isSubstituted when quantityCanceled #quantityCanceled and substitutionItem #substitutionItem"() {
+        given:
+        domain.quantityCanceled = quantityCanceled
+        domain.substitutionItem = substitutionItem
+        domain.requisitionItems = []
+
+        when:
+        Boolean isItemSubstituted = domain.isSubstituted()
+
+        then:
+        isSubstituted == isItemSubstituted
+
+        where:
+        quantityCanceled | substitutionItem      || isSubstituted
+        0                | null                  || false
```

Comment:
probably also worth having a line where quantityCancelled is > 0 but isSubstituted is still false if there are no substitutionItem